hal/src/nRF52840/ota_flash_hal.cpp

```diff
@@ -794,7 +794,7 @@ void HAL_FLASH_Write_ServerAddress(const uint8_t *buf, bool udp)
 int HAL_Set_System_Config(hal_system_config_t config_item, const void* data, unsigned data_length)
 {
     unsigned offset = 0;
-    unsigned length = -1;
+    int length = -1;
```

Comment:
could we keep the `unsigned length` and just check for length > 0 below?

Reply:
Yeah, might be an option :+1:

Reply:
Fixed